### PR TITLE
Add non-SOCKS proxy support to dockerv2

### PR DESCRIPTION
**Pull request by rmsds**
* refactor setupHttpConn to use urlparse
* check noproxy environment variable
* handle http(s)_proxy variables if present

Signed-off-by: Ricardo Silva <ricardo.silva@epfl.ch>

### CODE REVIEW

File: imagegw/shifter_imagegw/dockerv2.py [hunk 1]
```diff
@@ -90,30 +102,62 @@ def _verify_manifest_signature(manifest, text, digest):
...
-        server = url
-    if ':' in server:
-        (server, portstr) = server.split(':', 1)
-        port = int(portstr)
-    if protocol == 'http':
-        conn = httplib.HTTPConnection(server, port=port)
-    elif protocol == 'https':
+
+    if target.scheme == 'http':
+        if 'http_proxy' in os.environ and needProxy(target.hostname):
```
Comment (scanon) [inline imagegw/shifter_imagegw/dockerv2.py]: Is needProxy a typo?  Should it beed need_proxy?

**Comment by scanon** [comment]
Could we add a test or two around this and maybe some documentation?

File: imagegw/shifter_imagegw/dockerv2.py (hunk 1 above)
Comment (rmsds) [inline imagegw/shifter_imagegw/dockerv2.py]: Indeed, I missed that one when I updated the code to match the new coding style.

**Comment by rmsds** [comment]
I'll fix the type and add a few tests.

**Comment by scanon** [comment]
I'm closing this out and working on merging with the dev branch which will eventually become the next release.

**Pull request closed without merging**